src/Nethermind/Nethermind.Blockchain/Spec/ChainHeadSpecProvider.cs

```diff
@@ -42,6 +45,26 @@ public ChainHeadSpecProvider(ISpecProvider specProvider, IBlockFinder blockFinde
 
         public long[] TransitionBlocks => _specProvider.TransitionBlocks;
         
-        public IReleaseSpec GetSpec() => GetSpec(_blockFinder.FindBestSuggestedHeader()?.Number ?? 0);
+        public IReleaseSpec GetSpec()
+        {
+            long headerNumber = _blockFinder.FindBestSuggestedHeader()?.Number ?? 0;
+            if (headerNumber == _lastHeader)
+            {
+                IReleaseSpec releaseSpec = _headerSpec;
+                if (headerNumber == _lastHeader)
```

Comment:
I don't know the functionality so I need some clarification :)
GetSpec returns the best cached spec. 
Does it have to be the latest? We don't have a monitor here, so it can change between ln54 and ln58. Which means we don't care if it's the latest. So if we don't care that it's the latest then we don't need that double check.